File: test/language/templates/cpp98/InstantiateImportedTest.cpp
#include "gtest/gtest.h"

#include "templates/instantiate_imported/InstantiateImported.h"

namespace templates
{
namespace instantiate_imported
{

TEST(InstantiateImportedTest, readWrite)
{
    InstantiateImported instantiateImported;
    pkg::U32 u32;
    u32.setValue(13);
    instantiateImported.setTest32(u32);
    Test_string str;
    str.setValue("test");
    instantiateImported.setTestStr(str);

    zserio::BitStreamWriter writer;
    instantiateImported.write(writer);
    size_t bufferSize = 0;
    const uint8_t* buffer = writer.getWriteBuffer(bufferSize);

    zserio::BitStreamReader reader(buffer, bufferSize);
    InstantiateImported readInstantiateImported(reader);

    ASSERT_TRUE(instantiateImported == readInstantiateImported);
}

} // namespace instantiate_imported
} // namespace templates

File: test/language/templates/java/templates/InstantiateImportedTest.java
package templates;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import zserio.runtime.io.BitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamWriter;

import templates.instantiate_imported.InstantiateImported;
import templates.instantiate_imported.pkg.U32;
import templates.instantiate_imported.Test_string;

public class InstantiateImportedTest
{
    @Test
    public void readWrite() throws IOException
    {
        final InstantiateImported instantiateImported = new InstantiateImported();
        instantiateImported.setTest32(new U32(13));
        instantiateImported.setTestStr(new Test_string("test"));

        final ByteArrayBitStreamWriter writer = new ByteArrayBitStreamWriter();
        instantiateImported.write(writer);

        final BitStreamReader reader = new ByteArrayBitStreamReader(
                writer.toByteArray(), writer.getBitPosition());
        final InstantiateImported readInstantiateImported = new InstantiateImported(reader);
        assertTrue(instantiateImported.equals(readInstantiateImported));
    }
}
